server/api/app/Applications/User/Controllers/UserController.php

```diff
@@ -156,4 +158,42 @@ public function updateMyProfile(MyProfile $request)
     {
         $this->userService->updateMyProfile($request);
     }
+
+    /**
+     * Handle the avatar upload for the authenticated user.
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function uploadAvatar(Request $request): JsonResponse
+    {
+        try {
+            $userId = Route::current()->parameter('id');
+            $authenticatedUser = Auth::user();
+
+            $userDTO = $this->userService->uploadAvatar($userId, $request, $authenticatedUser);
+
+            return response()->json($userDTO, 200);
+        } catch (ValidationException $e) {
+            return response()->json([
+                'message' => 'Validation error.',
+                'errors' => $e->errors(),
+            ], 422);
+        } catch (\Symfony\Component\HttpKernel\Exception\NotFoundHttpException $e) {
+            return response()->json([
+                'message' => 'User not found.',
+            ], 404);
+        } catch (AuthorizationException $e) {
+            return response()->json([
+                'message' => $e->getMessage(),
```

Comment:
This can be circumvented by using can:uploadAvatar/upoadImage/modufyUserData gate/middleware for the route/ route group